### PR TITLE
Fixing slf4j dependency version in azure-cosmos-spark

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description

slf4j isn't following semantic versioning - so the conflicting slf4j versions (1.7.36 references in azure-cosmos-spark) and 1.7.30 used in Spark 3.1 and Spark 3.2 can result in the following error when customers use the Cosmos Spark connector + a custom jar using slf4j 1.7.30 (same as provided in Spark runtime)

```
Caused by: java.lang.LinkageError: loader constraint violation: when resolving method "org.slf4j.impl.StaticLoggerBinder.getLoggerFactory()Lorg/slf4j/ILoggerFactory;" the class loader (instance of org/apache/spark/util/ChildFirstURLClassLoader) of the current class, org/slf4j/LoggerFactory, and the class loader (instance of sun/misc/Launcher$AppClassLoader) for the method's defining class, org/slf4j/impl/StaticLoggerBinder, have different Class objects for the type org/slf4j/ILoggerFactory used in the signature
  at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:423)
  at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:362)
  at com.azure.cosmos.spark.diagnostics.BasicLoggingTrait.com$azure$cosmos$spark$diagnostics$BasicLoggingTrait$$log(BasicLoggingTrait.scala:10)
  at com.azure.cosmos.spark.diagnostics.BasicLoggingTrait.com$azure$cosmos$spark$diagnostics$BasicLoggingTrait$$log$(BasicLoggingTrait.scala:10)
  at com.azure.cosmos.spark.CosmosItemsDataSource.com$azure$cosmos$spark$diagnostics$BasicLoggingTrait$$log$lzycompute(CosmosItemsDataSource.scala:23)
  at com.azure.cosmos.spark.CosmosItemsDataSource.com$azure$cosmos$spark$diagnostics$BasicLoggingTrait$$log(CosmosItemsDataSource.scala:23)
  at com.azure.cosmos.spark.diagnostics.BasicLoggingTrait.logInfo(BasicLoggingTrait.scala:20)
  at com.azure.cosmos.spark.diagnostics.BasicLoggingTrait.logInfo$(BasicLoggingTrait.scala:19)
  at com.azure.cosmos.spark.CosmosItemsDataSource.logInfo(CosmosItemsDataSource.scala:23)
  at com.azure.cosmos.spark.CosmosItemsDataSource.(CosmosItemsDataSource.scala:24)
  at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
  at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
  at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
  at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
  at java.lang.Class.newInstance(Class.java:442)
  at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:380)
  ... 95 more
```

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3_2-12/pom.xml [hunk 1]
```diff
@@ -98,7 +98,7 @@
     <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-simple</artifactId>
-      <version>1.7.36</version> <!-- {x-version-update;org.slf4j:slf4j-simple;external_dependency} -->
+      <version>1.7.30</version> <!-- {x-version-update;cosmos_org.slf4j:slf4j-simple;external_dependency} -->
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/pom.xml:101]: Wouldn't this will cause conflicts with azure-cosmos shading?

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/pom.xml:101]: /azp run java - cosmos - ci

**Comment by FabianMeiswinkel** [comment]
slf4j version wasn't the culprit - so changes not necessary...

**Pull request closed without merging**